src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -47,10 +61,131 @@ public partial class MediaManager : Java.Lang.Object, IPlayer.IListener
+	}
+
+	/// <summary>
+	/// Method checks for required Permission for Android Notifications and requests them if needed
+	/// </summary>
+	/// <returns></returns>
+	static async Task CheckAndRequestForeGroundPermission(CancellationToken cancellationToken)
+	{
+		var status = await Permissions.CheckStatusAsync<AndroidPermissions>().WaitAsync(cancellationToken).ConfigureAwait(false);
+		if (status == PermissionStatus.Granted)
+		{
+			return;
+		}
+		else if (status == PermissionStatus.Denied && Build.VERSION.SdkInt > BuildVersionCodes.O)
+		{
+			await Shell.Current.DisplayAlert("Permission Required", "Notification permission is required to show media controls.", "Ok");
+		}
+		await Permissions.RequestAsync<AndroidPermissions>().WaitAsync(cancellationToken).ConfigureAwait(false);
+	}
+	
+	async Task StartService(CancellationToken cancellationToken)
+	{
+		var bitmap = await GetBitmapFromUrl(MediaElement.MetaDataArtworkUrl, Platform.AppContext.Resources, cancellationToken);
+		var mediaMetaData = new MediaMetadataCompat.Builder();
+		mediaMetaData.PutString(MediaMetadataCompat.MetadataKeyArtist, MediaElement.MetaDataArtist);
+		mediaMetaData.PutString(MediaMetadataCompat.MetadataKeyTitle, MediaElement.MetaDataTitle);
+		mediaMetaData.PutBitmap(MediaMetadataCompat.MetadataKeyAlbumArt, bitmap);
+		mediaMetaData.PutLong(MediaMetadataCompat.MetadataKeyDuration, Player?.Duration ?? 0);
+		mediaMetaData.Build();
+		
+		mediaSession?.SetMetadata(mediaMetaData.Build());
+		var intent = new Intent(Android.App.Application.Context, typeof(MediaControlsService));
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.O)
+		{
+			intent.PutExtra("title", MediaElement.MetaDataTitle);
+			intent.PutExtra("artist", MediaElement.MetaDataArtist);
+			intent.PutExtra("albumArtUri", MediaElement.MetaDataArtworkUrl);
+			intent.PutExtra("position", ((long)MediaElement.Position.TotalSeconds));
+			intent.PutExtra("currentTime", SystemClock.ElapsedRealtime());
+			intent.PutExtra("duration", ((long)MediaElement.Duration.TotalSeconds));
+		}
+		
+		intent.PutExtra("token", mediaSession?.SessionToken);
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.O)
+		{
+			Android.App.Application.Context.StartForegroundService(intent);
+		}
+		else
+		{
+			Android.App.Application.Context.StartService(intent);
+		}
+	}
+	
+	void BroadcastUpdate(string action)
+	{
+		if (Build.VERSION.SdkInt >= BuildVersionCodes.Tiramisu)
+		{
+			Logger?.LogError("{LocalBroadcastManager} not supported on Android 13 and above.", typeof(LocalBroadcastManager));
+			return;
+		}
+		Intent intent = new Intent(MediaControlsService.ACTION_UPDATE_UI);
+		intent.PutExtra("ACTION", action);
+		LocalBroadcastManager.GetInstance(Platform.AppContext).SendBroadcast(intent);
+	}
+	
+	public static async Task<Bitmap?> GetBitmapFromUrl(string? url, Resources? resources, CancellationToken cancellationToken)
```

Comment:
Since this method is public we should make CancellationToken optional, to follow the BCL pattern.
```suggestion
	public static async Task<Bitmap?> GetBitmapFromUrl(string? url, Resources? resources, CancellationToken cancellationToken = default)
```